File: Semestr II/Metody Programowania/Prolog/Lista 3/zad2.pl
exp(X, Y, Z):-
  Z is X^Y.
%-----------------------------------
/*
  count(Elem,List, Count):-
  select(Elem,List,X),
  Count is Count + 1,
  count(Elem,X,Count).
*/
count(_,[],0).
count(X, [X|T], N):-
  count(X,T,C),!,
  N is C + 1.
count(X,[_|T],N):-
  count(X,T,N).
%-----------------------------------
filter([],[]).
filter([H|T],[H|S]):-
  H >= 0,
   !,
  filter(T,S).
filter([H|T],S):-
  H < 0,
  filter(T,S).

File: SemestrIV/DatabaseApplication/Lista1/zad2.rb
require 'awesome_print'
require 'tiny_tds'
server = 'kapbd.database.windows.net'
database = 'Kurs'
username = 'mbronek7@kapbd'
password = 'zaq1@WSX'
client = TinyTds::Client.new username: username, password: password, 
    host: server, port: 1433, database: database, azure: true

puts "\nZad2 \n \n"

tsql =
"
SELECT SalesLT.ProductModel.Name AS ProductName, COUNT(SalesLT.Product.ProductModelID) AS Quantity
FROM  SalesLT.ProductModel
INNER JOIN SalesLT.Product ON SalesLT.Product.ProductModelID = SalesLT.ProductModel.ProductModelID
GROUP BY SalesLT.ProductModel.Name
HAVING COUNT(SalesLT.Product.ProductModelID) > 1;
"
result = client.execute(tsql)
result.each do |row|
    puts  row
end


